tests/test_serialization.py

```diff
@@ -0,0 +1,154 @@
+from typing import Optional, Type
+
+import attrs
+import pytest
+
+from web_poet import (
+    HttpResponse,
+    HttpResponseBody,
+    Injectable,
+    RequestUrl,
+    ResponseUrl,
+    WebPage,
+)
+from web_poet.page_inputs.url import _Url
+from web_poet.serialization import (
+    SerializedData,
+    deserialize,
+    read_serialized_data,
+    register_serialization,
+    serialize,
+    write_serialized_data,
+)
+
+
+def _compare_webpages(p1: WebPage, p2: WebPage) -> None:
```

Comment:
WebPage has `eq=False`, also `.url` is not comparable anyway.

Reply:
It's common to name such functions as asserts: assert_pages_equal. With `compare` it's not clear if it returns something, or raises an exception.